Check overwrite and destinationCollection exist during Move action

odrix opened the pull request:
If overwrite is false and destination exist 
then return precondition failed (412).

With precondition failed (412) Windows ask user for overwrite or not.

ramondeklein commented:
Thank you for you pull-request. You are right that we should return `PreconditionFailed` when the item exists and overwrite isn't set. Currently `Forbidden` is returned which is not according to the specification.

I doubt if we should check if the item exists before overwriting. This requires an additional check upfront. We could also choose to leave it to the store, which has more knowledge of the underlying storage and might be more efficient. Would you think that it would suffice to change `DiskStoreCollection.cs` and change line 318 to return `PreconditionFailed`? We then return a `MultiStatus` code, where the details hold the specific failure.

What do you think?

odrix commented:
We have implemented a _CustomStoreCollection_ inspired of _DiskStoreCollection_.
We have try to change _Forbidden_ to _PreconditionFailed_, but the return status is wrapped into a _MultiStatus_ response then Windows file explorer believe it is ok but it is not and it never ask user for overwrite.
That why i write this pull-reqiest.
For our case, we have implemented our custom MoveHandler, waiting this pull-request is ok.

ramondeklein commented:
Just wanted to check. I will merge this pull-request and push a new version to NuGet the following days.

ramondeklein commented:
Version 0.1.26.0 has been released and pushed to NuGet. It should be available in a few hours...

odrix commented:
Thanks again
I will watch tomorrow
